runtime/near-vm-logic/src/logic.rs

```diff
@@ -13,6 +13,105 @@ use std::mem::size_of;
 
 type Result<T> = ::std::result::Result<T, HostErrorOrStorageError>;
 
+pub struct GasCounter<'a> {
+    /// The amount of gas that was irreversibly used for contract execution.
+    burnt_gas: Gas,
+    /// `burnt_gas` + gas that was attached to the promises.
+    used_gas: Gas,
+    config: &'a Config,
+    prepaid_gas: Gas,
+    is_view: bool,
+}
+
+impl<'a> GasCounter<'a> {
+    pub fn new(config: &'a Config, prepaid_gas: Gas, is_view: bool) -> Self {
+        Self { burnt_gas: 0, used_gas: 0, config: config, prepaid_gas, is_view }
+    }
+    pub fn deduct_gas(&mut self, burn_gas: Gas, use_gas: Gas) -> Result<()> {
```

Comment:
I cannot visually diff the `deduct_gas` and other functions of `GasCounter` but I assume they were moved from `Logic` without change?

Reply:
> but I assume they were moved from Logic without change?
yeah, only